api_spec.yaml

```diff
@@ -107,8 +107,98 @@ paths:
         
         '500':
           $ref: '#/components/responses/500'
+          
+    get:
+      tags:
+        - OAuth2 Recipe
+      operationId: getOAuth2AuthorizationCode
+      description: |
+        get authorization code details belonging to a client.
+        [Click here](https://app.code2flow.com/gq43eMfvLwfg) for flow diagram.
+      parameters:
```

Comment:
make a list of all the plugin interface functions and reply to this comment (or put them in the master checklist). The functions should have the full function signature for each function.

Reply:
list of plugin interface methods:

- [ ]  getOAuth2ClientById(AppIdentifier appIdentifier, String clientId) throws StorageQueryException, UnknownOAuth2ClientIdException
- [ ]  removeOAuth2Client(AppIdentifier appIdentifier, String clientId) throws StorageQueryException, UnknownOAuth2ClientIdException
- [ ]  getOAuth2ClientScopes(AppIdentifier appIdentifier, String clientId) throws StorageQueryException, UnknownOAuth2ClientIdException, UnknownOAuth2ScopeException
- [ ]  removeOAuth2ClientScope(AppIdentifier appIdentifier, String clientId, String scope) throws StorageQueryException, UnknownOAuth2ClientIdException, UnknownOAuth2ScopeException
- [ ]  createOAuth2Scope(AppIdentifier appIdentifier, String scope) throws StorageQueryException, DuplicateOAuth2ScopeException, TenantOrAppNotFoundException
- [ ]  updateOAuth2Scope(AppIdentifier appIdentifier, String scope, String newName) throws StorageQueryException, DuplicateOAuth2ScopeException
- [ ]  removeOAuth2Scope(AppIdentifier appIdentifier, String scope) throws StorageQueryException, DuplicateOAuth2ScopeException
- [ ]  getOAuth2Scopes(AppIdentifier appIdentifier) throws StorageQueryException
- [ ]  createOAuth2AuthorizationCode(TenantIdentifier tenantIdentifier, String codeHash, String sessionHandle, String clientId, long createdAtMs, long expiresAtMs, List<String> scope, String redirectUri, String accessType, String codeChallenge, String codeChallengeMethod) throws StorageQueryException, DuplicateOAuth2AuthorizationCodeHash
- [ ]  getOAuth2TokenInfoByAccessTokenHash(TenantIdentifier tenantIdentifier, String accessTokenHash) throws StorageQueryException, UnknownOAuth2AccessTokenHashHashException
- [ ]  updateOAuth2TokenInfoByRefreshTokenHash(TenantIdentifier tenantIdentifier, String refreshTokenHash, String newAccessTokenHash, String newRefreshTokenHash, long accessTokenExpiresAtMs, Long refreshTokenHashExpiresAtMs, long updatedAtMs) throws StorageQueryException, UnknownOAuth2RefreshTokenHashHashException, DuplicateOAuth2RefreshTokenHash, DuplicateOAuth2AccessTokenHash
- [ ]  removeOAuth2TokensExpiredBefore(long now) throws StorageQueryException
- [ ]  removeOAuth2TokensBySessionHandle(TenantIdentifier tenantIdentifier, String sessionHandle) throws StorageQueryException
- [ ]  removeOAuth2TokenByAccessTokenHash(TenantIdentifier tenantIdentifier, String accessTokenHash) throws StorageQueryException
- [ ]  removeOAuth2TokensByClientId(TenantIdentifier tenantIdentifier, String accessTokenHash) throws StorageQueryException
- [ ]  removeOAuth2AuthorizationCodesExpiredBefore(long now) throws StorageQueryException
- [ ]  removeOAuth2AuthorizationCodesByClientId(long now) throws StorageQueryException
- [ ]  removeOAuth2AuthorizationCodesBySessionHandle(long now) throws StorageQueryException

- [ ]  void createOAuth2Client_Transaction(AppIdentifier appIdentifier, TransactionConnection con, OAuth2Client oAuth2Client)    throws StorageQueryException, DuplicateOAuth2ClientSecretHash, DuplicateOAuth2ClientIdException,DuplicateOAuth2ClientNameException, TenantOrAppNotFoundException;
- [ ]     void updateOAuth2Client_Transaction(AppIdentifier appIdentifier, TransactionConnection con, String clientId, String name,    String clientSecretHash, List<String> redirectUris, long updatedAtMs)           throws StorageQueryException, UnknownOAuth2ClientIdException, DuplicateOAuth2ClientSecretHash;
- [ ]     OAuth2Client getOAuth2ClientByIdAndClientSecretHash_Transaction(AppIdentifier appIdentifier, TransactionConnection con, String clientId, String clientSecretHash)      throws StorageQueryException, UnknownOAuth2ClientIdOrSecretException;
- [ ]     void insertOAuth2ClientScope_Transaction(AppIdentifier appIdentifier, TransactionConnection con, String clientId, String scope, boolean requiresConsent) throws StorageQueryException, UnknownOAuth2ClientIdException, UnknownOAuth2ScopeException, DuplicateOAuth2ClientScopeException;

- [ ]     OAuth2AuthorizationCode getOAuth2AuthorizationCodeByCodeHash_Transaction(TenantIdentifier tenantIdentifier, TransactionConnection con, String codeHash)     throws StorageQueryException, UnknownOAuth2AuthorizationCodeHashException;
- [ ]     void removeOAuth2AuthorizationCodeByCodeHash_Transaction(TenantIdentifier tenantIdentifier, TransactionConnection con, String codeHash)   throws StorageQueryException, UnknownOAuth2AuthorizationCodeHashException;

- [ ]     void createOAuth2Token_Transaction(TenantIdentifier tenantIdentifier, TransactionConnection con, String clientId,String sessionHandle, List<String> scope, String accessTokenHash,    String refreshTokenHash, long createdAtMs, long accessTokenExpiresAtMs,   Long refreshTokenHashExpiresAtMs)   throws StorageQueryException, UnknownOAuth2ClientIdException;